### PR TITLE
Fix `flutter update-packages` regression by fixing parameters in "pub get" runner

### PR DESCRIPTION
**Pull request by nehalvpatel**
With the update to the "pub get" runner in https://github.com/flutter/flutter/pull/115801, there were a couple parameters (printProgress and retry) that were left out. This causes `flutter update-packages` to print unexpected output as seen in https://github.com/flutter/flutter/issues/116627.

This PR fixes that by hooking up the parameters.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
Code LGTM. Your CI tests are failing from an infra issue, and I think you'll need to add tests.

**Comment by christopherfujino** [comment]
cc @andrewkolos

**Comment by nehalvpatel** [comment]
Looks like there's something in this PR that causes all of the following tests to time out. They're all Windows tests. It's specifically on the "download dependencies" step.

**Example of success** from a different PR: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8794762758857728449/+/u/download_dependencies/stdout

**Example of failure**: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8794777666714712193/+/u/download_dependencies/stdout
- The last line is from the infra tooling automatically cancelling the task after it stalled for 3 hours.

---

I believe it's this last line in `flutter.bat` where it gets stuck: https://github.com/flutter/flutter/blob/b84bfa35264385a897e6d85d039eae0ea6ba90ab/bin/flutter.bat#L55

---

The tests:

- [Windows build_tests_1_3](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098194567)
- [Windows build_tests_2_3](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098194703)
- [Windows build_tests_3_3](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098194842)
- [Windows customer_testing](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098195012)
- [Windows framework_tests_libraries](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098195146)
- [Windows framework_tests_misc](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098195308)
- [Windows framework_tests_widgets](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098195458)
- [Windows tool_integration_tests_1_6](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098196193)
- [Windows tool_integration_tests_2_6](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098196324)
- [Windows tool_integration_tests_3_6](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098196437)
- [Windows tool_integration_tests_4_6](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098196571)
- [Windows tool_integration_tests_5_6](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098196724)
- [Windows tool_integration_tests_6_6](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098196876)
- [Windows tool_tests_commands](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098196980)
- [Windows tool_tests_general](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098197119)
- [Windows web_tool_tests_1_2](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098197232)
- [Windows web_tool_tests_2_2](https://github.com/flutter/flutter/pull/116687/checks?check_run_id=10098197346)

**Comment by christopherfujino** [comment]
```
Checking Dart SDK version... 
Downloading Dart SDK from Flutter engine ... 
Expanding downloaded archive...
Building flutter tool... 
Running pub upgrade... 
Downloading Material fonts...                                      510ms
Downloading Gradle Wrapper...                                       12ms
Downloading package sky_engine...                                  116ms
Downloading flutter_patched_sdk tools...                           305ms
Downloading flutter_patched_sdk_product tools...                   300ms
Downloading windows-x64 tools...                                 2,247ms
Downloading windows-x64/font-subset tools...                       135ms
^CTerminate batch job (Y/N)? 
```

So what's happening here is that LUCI is trying to run `flutter update-packages`. In order, what happens is:

1. the `flutter.bat` batch script downloads the Dart SDK
2. the `flutter.bat` script compiles the flutter tool
3. the `flutter.bat` script runs `dart pub upgrade` in the flutter_tools directory
4. the `flutter.bat` script invokes the flutter tool with `update-packages` as the sub-command
5. the tool starts caching pre-built binaries required by the `update-packages` sub-command (does this finish and the next command start?)
6. the tool hangs until LUCI times it out after 3 hours

Given the logs in the succeeding step you linked, I'm guessing step 5 finished. I'm not sure exactly what happened, I think you need to investigate more, but maybe your change made the tool stop logging "Running pub upgrade...", then invoked pub. Then, for whatever reason, it's retrying forever until it gets timed out?

One thing you could do is hack the tool to always run this sub-command in verbose mode just for your PR, push it to github, and then look at the verbose LUCI logs.

**Comment by nehalvpatel** [comment]
Thanks for taking a look! I'll give that a shot.

**Comment by nehalvpatel** [comment]
@christopherfujino

Looks like the issue was fixed.

Apparently it was because I was using ProcessUtils.start for the non-output mode, which I now see does warn you in the documentation to make sure to read all the data from the stdout / stderr streams, but as of now only trips up on Windows if you don't do that. I left a comment about it in pub.dart. 😵‍💫

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 1]
```diff
@@ -393,45 +397,72 @@ class _DefaultPub implements Pub {
...
-        final StreamSubscription<List<int>> stdoutSubscription =
-          process.stdout.listen(stdio.stdout.add);
-        final StreamSubscription<List<int>> stderrSubscription =
-          process.stderr.listen(stdio.stderr.add);
-
-        await Future.wait<void>(<Future<void>>[
-          stdoutSubscription.asFuture<void>(),
-          stderrSubscription.asFuture<void>(),
-        ]);
+    if (!retry) {
```
Comment (sigurdm) [inline packages/flutter_tools/lib/src/dart/pub.dart:406]: Are we sure we need this?

If I'm correct retry is only set for the `--offline` mode, and pub itself will never retry enything in offline mode (as it won't make any requests)

Comment (nehalvpatel) [reply packages/flutter_tools/lib/src/dart/pub.dart:406]: Good catch, yea, I was not sure about this. I will remove it.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**